scripts/solve_network.py

```diff
@@ -762,6 +762,37 @@ def add_pipe_retrofit_constraint(n):
     n.model.add_constraints(lhs == rhs, name="Link-pipe_retrofit")
 
 
+def add_geothermal_chp_constraint(n):
+    elec_index = n.links.loc[
+        n.links.carrier == "geothermal organic rankine cycle"
+    ].index
+    heat_index = n.links.loc[n.links.carrier == "geothermal heat district heat"].index
+
+    p_nom_lhs = (
+        n.model["Link-p_nom"].loc[heat_index] - n.model["Link-p_nom"].loc[elec_index]
+    )
+
+    n.model.add_constraints(
+        p_nom_lhs == 0,
+        name="equalizes_p_nom_of_chp_elec_and_chp_district_heat",
+    )
+
+
+def add_flexible_egs_constraint(n):
+    well_index = n.links.loc[n.links.carrier == "geothermal heat"].index
+    storage_index = n.storage_units.loc[
+        n.storage_units.carrier == "geothermal heat  "
```

Comment:
looks like it actually should be without the space `n.storage_units.carrier == "geothermal heat"`

Reply:
Good catch, thanks!